### PR TITLE
chore: allow urllib3>=1.25.10

### PR DESCRIPTION
**Pull request by JohnVillalovos**
Allow use of urllib3>=1.25.10. By default it will use urllib3>=2 as that is the latest version. But this will allow people to pin urllib3<2 and still use the latest version of requests and responses.

Closes: #657

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 install_requires = [
     "requests>=2.30.0,<3.0",
-    "urllib3>=2.0.0,<3.0",
+    "urllib3>=1.25.10",
```
Comment (markstory) [inline setup.py:21]:
```suggestion
    "urllib3>=1.25.10,<3.0",
```

We should keep the upper bound. Future versions of urllib3 may require changes in responses.

Comment (JohnVillalovos) [reply setup.py]: Thanks. Done.

**Comment by beliaev-maksim** [comment]
please add a matrix of tests in CI. We need to test on both 1.x and 2.x

otherwise we give a promise that we cannot guarantee

**Comment by JohnVillalovos** [comment]
> please add a matrix of tests in CI. We need to test on both 1.x and 2.x
> 
> otherwise we give a promise that we cannot guarantee

Thanks.

Done, I think. The CI doesn't run for me yet so I'm not 100% certain.

**Comment by JohnVillalovos** [comment]
@beliaev-maksim or @markstory   Can one of you enable the workflow to run? Thanks.

**Comment by JohnVillalovos** [comment]
> Done, I think. The CI doesn't run for me yet so I'm not 100% certain.

I was able to run the CI over in my fork. And I got the tests to pass. https://github.com/JohnVillalovos/responses/pull/2

**Review by Seluj78 — APPROVED** [review]

**Pull request merged**